app/src/main/java/nodomain/freeyourgadget/gadgetbridge/util/CSVExport.java

```diff
@@ -0,0 +1,51 @@
+package nodomain.freeyourgadget.gadgetbridge.util;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+
+import nodomain.freeyourgadget.gadgetbridge.devices.AbstractSampleProvider;
+import nodomain.freeyourgadget.gadgetbridge.entities.AbstractActivitySample;
+
+/**
+ * Created by Vebryn on 29/05/17.
+ */
+
+public class CSVExport {
+    private static final Logger LOG = LoggerFactory.getLogger(AbstractSampleProvider.class);
+
+    public static void exportToCSV(AbstractActivitySample[] activitySamples, File outFile) {
```

Comment:
I think it's sufficient to use `ActivitySample` here instead of `AbstractActivitySample`.